torch/_refs/nn/functional/__init__.py

```diff
@@ -0,0 +1,48 @@
+import torch
+
+import torch._prims.utils as utils
+from torch._prims.utils import TensorLikeType, TensorLike, NumberType, Number
+import torch._refs as refs
+
+from typing import Optional
+
+all = [
+    "elu",
+]
+
+# elu is implemented specially because it has an alpha argument
+def elu(a: TensorLikeType, alpha: Optional[NumberType] = None, inplace: bool = False):
+    """
+    Reference implementation of torch.nn.functional.elu
+    """
+
+    if inplace:
+        raise NotImplementedError
+
+    # Type checks
+    assert isinstance(a, TensorLike)
+    assert alpha is None or isinstance(alpha, Number)
+
+    computation_dtype, result_dtype = refs._elementwise_dtypes(
+        a, type_promotion=refs.ELEMENTWISE_TYPE_PROMOTION_KIND.OP_MATH
+    )
+
+    (a,) = refs._convert_dtype(a, dtype=computation_dtype)
```

Comment:
we probably should use opmath decorator like decompositions in _decomps

Reply:
one trickiness is for the decorator to be completely sound, we need to know a bit about the operators. It's safer to explicitly feed the operator some information about the argument types instead of sniffing it out by just looking and seeing what the user provided and doing some different if it's a tensor vs number.